_posts/2018-03-08-reclaiming-disk-space-from-mongodb.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Reclaiming Disk Space from MongoDB
+author: David Davis
+---
+MongoDB can often eat up large amounts of disk space especially in deployments that see heavy
+loads. It's possible to reclaim some of this disk space by using Mongo's repair utility. Before you
+start this process though, **it is highly recommended you back up your MongoDB database**.
+
+First, check the amount of disk space being used by MongoDB by connecting to MongoDB and running the
+db stats command:
+
+```
+$ mongo pulp_database
+MongoDB shell version: 3.2.12
+connecting to: pulp_database
+> db.stats()
+{
+        "db" : "pulp_database",
+        "collections" : 44,
+        "objects" : 698,
+        "avgObjSize" : 639.756446991404,
+        "dataSize" : 446550,
+        "storageSize" : 811008,
+        "numExtents" : 0,
+        "indexes" : 147,
+        "indexSize" : 1474560,
+        "ok" : 1
+}
+```
+
+Here you can see my database is using only about 800 kilobytes as this is a new installation of
+Pulp. Now the command we want to run to reclaim disk space is [the repairDatabase
+command](https://docs.mongodb.com/manual/reference/command/repairDatabase/).
+
+There are a couple important things to note though. First, depending on the size of your database,
+this could take a long time. The second is that you'll need free space on your disk equal to the
```

Comment:
I am going to refrain from further suggestions - but my personal preference is to avoid 'you' and 'your database, pulp services' etc